server/client.go

```diff
@@ -1897,12 +1897,28 @@ func (c *client) flushSignal() {
 // Traces a message.
 // Will NOT check if tracing is enabled, does NOT need the client lock.
 func (c *client) traceMsg(msg []byte) {
-	maxTrace := c.srv.getOpts().MaxTracedMsgLen
-	if maxTrace > 0 && (len(msg)-LEN_CR_LF) > maxTrace {
+	opts := c.srv.getOpts()
+	maxTrace := opts.MaxTracedMsgLen
+	headersOnly := opts.TraceHeaders
+	suffix := LEN_CR_LF
+
+	// If TraceHeaders is enabled, extract only the header portion of the msg.
+	// The suffix is zeroed since the header is only being included.
+	if headersOnly {
+		msg, _ = c.msgParts(msg)
```

Comment:
`msg` here, will be actually the headers, but they still end with a `\r\n`.

Reply:
Understood, I did this intentionally. I will add a comment so its clear.